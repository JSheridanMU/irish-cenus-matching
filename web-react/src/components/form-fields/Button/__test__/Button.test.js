import * as React from 'react'
import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Button from '.././Button'

afterEach(cleanup)

test('renders without crashing', () => {
  render(<Button text={'Test Button'} />)
})

test('renders button with text', () => {
  const { getByText } = render(<Button text={'Test Button'} />)

  getByText('Test Button')
})

test('matches snapshot', () => {
  const tree = renderer.create(<Button text={'Test Button'} />).toJSON()
  expect(tree).toMatchSnapshot()
})
