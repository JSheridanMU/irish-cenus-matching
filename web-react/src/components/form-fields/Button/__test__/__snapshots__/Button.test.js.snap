// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root makeStyles-root-5 MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label makeStyles-label-6"
  >
    Test Button
  </span>
</button>
`;
